ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/LedgerApiBenchTool.scala

```diff
@@ -207,39 +118,214 @@ object LedgerApiBenchTool {
 
       for {
         _ <- regularUserSetupStep(adminServices)
-        submissionStepProducts <- submissionStep(
-          regularUserServices = regularUserServices,
-          adminServices = adminServices,
-          submissionConfig = config.workflow.submission,
-          metricRegistry = metricRegistry,
-        )
-        streams = config.workflow.streams.map(
-          ConfigEnricher.enrichStreamConfig(_, submissionStepProducts.map(_._2))
+        submissionStepResultO: Option[SubmissionStepResult] <- {
+          config.workflow.submission match {
+            case None =>
+              logger.info(s"No submission defined. Skipping.")
+              Future.successful(None)
+            case Some(submissionConfig) =>
+              submissionStep(
+                regularUserServices = regularUserServices,
+                adminServices = adminServices,
+                submissionConfig = submissionConfig,
+                metricRegistry = metricRegistry,
+              ).map(Some(_))
+          }
+        }
+
+        updatedStreamConfigs = config.workflow.streams.map(streamsConfig =>
+          ConfigEnricher.enrichStreamConfig(streamsConfig, submissionStepResultO)
         )
+
         _ = logger.info(
-          s"Stream configs adapted after the submission step: ${prettyPrint(streams)}"
+          s"Stream configs adapted after the submission step: ${prettyPrint(updatedStreamConfigs)}"
         )
         benchmarkResult <-
           if (config.latencyTest) {
```

Comment:
When I first read this it seemed to me we had the wrong abstraction.

Reply:
Discussed offline